Fixes/lab 04 (#62)
VirtuslabWorkshops/cloudyna-workshop: lab04/README.md

## Objectives

> - Import existing resources to bind them into Terraform `state`

Reviewer: `bind with` seems consistent with a later occurrence
```suggestion
- Import existing resources to bind them with the Terraform `state`
```

- Inspect `state` 
- Add rule to ignore certain changes in objects

## Terraform state

Terraform keeps information about effect of its work in `state`. 
In this scenario you will create resources manually and then map them with configuration files.
After that you will remove resource from `state` and apply configuration.

Key points:
- effectively this is text file which helds information about objects managed by Terraform
- `state` is what Terraform _believes_ is out there, it is being used to compare expected vs existing state
- terraform allows to bind existing resource with state using import command
Terraform keeps information about effect of its work in file `terraform.tfstate` often referred as`state` or `state file`. 
In this scenario we will create resources manually and then import them into Terraform state.
After that we will remove resource from `state` and apply configuration.

### Importing resources
   
1. Create sample resources using bash script
    - inspect [/scripts/createRGSA.sh](./scripts/createRGSA.sh) and put your initials in relevant places
    - run script
       ```bash
       cd scripts
       chmod +x createRGSA.sh

>        ./createRGSA.sh

Author: added command but there's no need to chmod the file`
`-rwxrwxr-x 1 mgolaszewski mgolaszewski 568 Dec  7 19:55 ../scripts/createRGSA.sh`

>        ./createRGSA.sh

Reviewer: it's not about the ability to run the script but to reduce cognitive load when there's a new thing to comprehend :)

> 2. Run `terraform plan` and compare execution plan with exising infrastructure on the azure portal. 

Reviewer: Why remove variable input? Maybe add a note about changing defaults in variables.tf? 
```suggestion
2. Run `terraform plan` and compare the execution plan with existing infrastructure on the Azure portal. 
```

Author: added note about variable imputs

> 6. See how terraform handles external changes. Add new tag to storage account in azure portal and re apply terraform.

Reviewer: ```suggestion
6. See how terraform handles external changes. Add a new tag to the storage account in the Azure portal and re-apply terraform.
```
